### PR TITLE
fix: migrate from tap to node test and c8

### PR DESCRIPTION
**Pull request by nimesh0505**
#### Checklist

This PR replaces tap with node:test for testing and adds c8 for coverage.

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,6 +31,7 @@
     "@fastify/pre-commit": "^2.1.0",
     "@fastify/type-provider-typebox": "^5.0.0-pre.fv5.1",
     "@types/node": "^22.0.0",
+    "c8": "^10.1.2",
```
Comment (simoneb) [inline package.json:34]: remove tap from dependencies

File: package.json [hunk 2]
```diff
@@ -8,7 +8,8 @@
   "scripts": {
     "lint": "standard",
     "test": "npm run test:unit && npm run test:typescript",
-    "test:unit": "tap",
+    "test:unit": "node --test",
```
Comment (gurgunday) [inline package.json:11]:
```suggestion
    "test:unit": "c8 --100 node --test",
```

**Review by gurgunday — CHANGES_REQUESTED** [review]
If the taprc file did not contain *disable-coverage: true*, please enable c8 coverage; for how, see my suggested change

File: test/bundlers.test.js [hunk 3]
```diff
@@ -1,38 +1,35 @@
 'use strict'
 
-const { test } = require('tap')
+const { test } = require('node:test')
 const fp = require('../plugin')
 
-test('webpack removes require.main.filename', (t) => {
+test('webpack removes require.main.filename', async (t) => {
```
Comment (jsumners) [inline test/bundlers.test.js:6]: Why have all of the test function been refactored to async functions?

Comment (nimesh0505) [reply test/bundlers.test.js]: @jsumners Thank you for your question. You are right to notice that all test functions have been converted to async functions. While it's not strictly necessary for the migration from `tap` to `node:test`, I chose to do so due to these reasons:

1. Consistency - It provides a uniform pattern across all tests.
2. Future-proofing - It allows for easier integration of async operations in the future if needed.
3. Minimal impact: The change doesn't negatively affect the tests' functionality or performance.

However, if the team prefers to keep the tests as close to the original implementation as possible, we can certainly revert this change and only use async functions where necessary.

**Review by gurgunday — DISMISSED** [review]
lgtm

no strong opinions about https://github.com/fastify/fastify-plugin/pull/248#discussion_r1754961467

File: test/bundlers.test.js (hunk 3 above)
Comment (jsumners) [reply test/bundlers.test.js]: Unless there is a need for the functions to be async, please do not make them so.

**Comment by Uzlopak** [comment]
I agree with @jsumners . It doesnt make any sesnse and has no benefit of making functions async if there is nothing to be awaited etc.. So yes, please remove unnecesasry async.

**Comment by simoneb** [comment]
comments addressed

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**